client/src/http_default.rs

```diff
@@ -41,90 +41,64 @@ trait SetHeadersExt: Sized + SetSingleHeader {
     }
 }
 
-impl SetHeadersExt for AttoHttpRequestBuilderWithBytes {}
+impl SetHeadersExt for AttoHttpRequestBuilder {}
 
 impl SetHeadersExt for http::request::Builder {}
 
 /// Default request builder & sender implemented on top of `attohttpc` crate.
 ///
 /// Its main goal is not to be efficient, but simple. Its implementation contains
 /// some intermediate allocations that could be avoided with additional complexity.
-pub struct DefaultRequestBuilder {
-    method: Method,
-    url: String,
-    params: Option<Vec<(String, String)>>,
-    headers: Option<Headers>,
-    body: Option<Bytes>,
-}
+pub struct DefaultRequestBuilder(Result<AttoHttpRequestBuilder>);
 
 impl DefaultRequestBuilder {
+    fn and_then<F>(self, fun: F) -> Self
```

Comment:
`and_then` is a bit misleading as a method of this name usually accepts `Option<T>` or `Result<T, E>` and returns `Option<U>` or `Result<U, E>`

Reply:
Added a comment with explanation.